docs/keto/modeling/learn-the-concepts.mdx

```diff
@@ -0,0 +1,295 @@
+> editor of a document, and an editor is also a viewer of a document. Furthermore, documents can be put into a hierarchy of
+> folders, such that being able to view the parent folder implies being able to view the enclosed folders and documents.
+
+### Create a list of objects
+
+We start with the description and highlight all objects that we find:
+
+> We want to model `user` access to `documents` such that `users` can be an owner, editor, or viewer of a document. An owner is
+> also an editor of a `document`, and an editor is also a viewer of a `document`. Furthermore, `documents` can put into a
+> hierarchy of `folders`, where the ability to view the parent `folder` implies the ability to view the enclosed `folders` and
+> `documents`.
+
+We found these objects:
+
+- User
+- Document
+- Folder
+
+This list translates into our first version of the permission model.
+
+```ts title="permissions-v1.ts"
+import { Namespace, Context } from "@ory/keto-namespace-types"
+
+// highlight-start
+class User implements Namespace {}
+class Document implements Namespace {}
+class Folder implements Namespace {}
+// highlight-end
+```
+
+:::tip
+
+The objects and subjects from your description are [namespaces](../concepts/namespaces) in Ory Permissions, and are declared as
+classes in the Ory Permission Language. The convention is to spell these class names as TypeScript classes, capitalized and in
+singular form, for example `Document` instead of `documents`.
+
+:::
+
+### List relationships to other objects
+
+Read through the description sentence by sentence to find the list of relationships:
+
+> We want to model `user` access to `documents` such that `users` can be an `owner`, `editor`, or `viewer` of a `document`.
+
+<Mermaid
+  chart={`
+graph LR
+    Document -->|owner| User
+    Document -->|editor| User
+    Document -->|viewer| User
+`}
+/>
+
+> An owner is also an editor of a `document`, and an editor is also a viewer of a `document`. Furthermore, `documents` can put
+> into a hierarchy of `folders`, such that being able to view the parent `folder` implies being able to view the enclosed
+> `folders` and `documents`.
+
+<Mermaid
+  chart={`
+graph LR
+    Document -->|parent| Folder
+    Folder -->|parent| Folder
+    Folder -->|owner| User
+    Folder -->|editor| User
+    Folder -->|viewer| User
+`}
+/>
+
+Putting this all together in one graph:
+
+<Mermaid
+  chart={`
+graph LR
+    Document -->|owner| User
+    Document -->|editor| User
+    Document -->|viewer| User
+    Document -->|parent| Folder
+    Folder -->|parent| Folder
+    Folder -->|owner| User
+    Folder -->|editor| User
+    Folder -->|viewer| User
+`}
+/>
+
+### List permissions to check for each object
+
+When you perform an action on behalf of a user, you usually check a specific permission, such as `view` or `edit`, instead of a
+relationship, such as `owner`, and have the relationship implied by the permission.
```

Comment:
```suggestion
When you perform an action on behalf of a user, you often want to know what permissions the user has, for example `view` or `edit` rather than the relationship, for example `owner`. In this case you check the relationship of the user, which then implies the permissions this user has.
```
I misunderstood this the first time around, this should be more clear, wdyt

Reply:
I applied Patrik's suggestion, does this make things clearer already?